project-api/project-api-test/src/test/java/io/ballerina/projects/test/TestBuildProject.java

```diff
@@ -1531,6 +1531,63 @@ public void testAutoUpdateWithBuildFile() throws IOException {
         Assert.assertFalse(thirdBuildJson.isExpiredLastUpdateTime());
     }
 
+    @Test(description = "test auto updating dependencies with build file after removing Dependencies.toml")
+    public void testAutoUpdateWithBuildFileWithoutDepsToml() throws IOException {
+        Path projectPath = RESOURCE_DIRECTORY.resolve("myproject");
+        // Delete build file and Dependencies.toml file if already exists
+        Files.deleteIfExists(projectPath.resolve(TARGET_DIR_NAME).resolve(BUILD_FILE));
+        Files.deleteIfExists(projectPath.resolve(DEPENDENCIES_TOML));
+
+        // Set sticky false, to imitate the default build command behavior
+        BuildOptionsBuilder buildOptionsBuilder = new BuildOptionsBuilder();
+        buildOptionsBuilder.sticky(false);
+        BuildOptions buildOptions = buildOptionsBuilder.build();
+
+        // 1) Initialize the project instance
+        BuildProject project = null;
+        try {
+            project = TestUtils.loadBuildProject(projectPath, buildOptions);
+            project.save();
+        } catch (Exception e) {
+            Assert.fail(e.getMessage());
+        }
+        Assert.assertEquals(project.currentPackage().packageName().toString(), "myproject");
+        Path buildFile = project.sourceRoot().resolve(TARGET_DIR_NAME).resolve(BUILD_FILE);
+        Assert.assertTrue(buildFile.toFile().exists());
+        BuildJson initialBuildJson = readBuildJson(buildFile);
+        Assert.assertTrue(initialBuildJson.lastBuildTime() > 0);
```

Comment:
Shall we include specific messages to these assertions, so that when the test fails, we can clearly identify why from the message itself?

Reply:
Added some error messages at the needed places